test/units/galaxy/test_collection.py

```diff
@@ -350,43 +342,75 @@ def test_validate_certs_with_server_url(global_ignore_certs, monkeypatch):
-    get_plugin_options = MagicMock(side_effect=server_config)
-    monkeypatch.setattr(C.config, 'get_plugin_options', get_plugin_options)
-
+@pytest.mark.parametrize(
+    # False/True/omitted, True/omitted
+    "ignore_certs_cli,ignore_certs_cfg",
+    [
+        (None, None),
+        (None, True),
+        (None, False),
+        (True, None),
+        (True, True),
+        (True, False),
+    ]
+)
+def test_validate_certs_server_config(ignore_certs_cfg, ignore_certs_cli, monkeypatch):
+    server_names = ['server1', 'server2', 'server3']
+    cfg_lines = [
+        "[galaxy]",
+        "server_list=server1,server2,server3",
+        "[galaxy_server.server1]",
+        "url=https://galaxy.ansible.com/api/",
+        "validate_certs=False",
+        "[galaxy_server.server2]",
+        "url=https://galaxy.ansible.com/api/",
+        "validate_certs=True",
+        "[galaxy_server.server3]",
+        "url=https://galaxy.ansible.com/api/",
+    ]
     cli_args = [
         'ansible-galaxy',
         'collection',
         'install',
         'namespace.collection:1.0.0',
     ]
-    if global_ignore_certs:
+    if ignore_certs_cli:
         cli_args.append('--ignore-certs')
+    if ignore_certs_cfg is not None:
+        monkeypatch.setattr(C, 'GALAXY_IGNORE_CERTS', ignore_certs_cfg)
 
-    galaxy_cli = GalaxyCLI(args=cli_args)
-    mock_execute_install = MagicMock()
-    monkeypatch.setattr(galaxy_cli, '_execute_install_collection', mock_execute_install)
-    galaxy_cli.run()
+    with tempfile.NamedTemporaryFile(suffix='.cfg') as tmp_file:
+        tmp_file.write(to_bytes('\n'.join(cfg_lines), errors='surrogate_or_strict'))
+        tmp_file.flush()
+
+        with patch.object(C, 'GALAXY_SERVER_LIST', server_names):
+            with patch.object(C.config, '_config_file', tmp_file.name):
+                C.config._parse_config_file()
+
+                galaxy_cli = GalaxyCLI(args=cli_args)
+                mock_execute_install = MagicMock()
+                monkeypatch.setattr(galaxy_cli, '_execute_install_collection', mock_execute_install)
+                galaxy_cli.run()
 
-    # server cfg, so should match def above, if not specified so it should use default (true)
-    assert galaxy_cli.api_servers[0].validate_certs is server_config[0].get('validate_certs', True)
-    assert galaxy_cli.api_servers[1].validate_certs is server_config[1].get('validate_certs', True)
-    assert galaxy_cli.api_servers[2].validate_certs is server_config[2].get('validate_certs', True)
+    # (not) --ignore-certs > server's validate_certs > (not) GALAXY_IGNORE_CERTS > True
+    assert galaxy_cli.api_servers[0].validate_certs is False
+    assert galaxy_cli.api_servers[1].validate_certs is (False if ignore_certs_cli else True)
+    assert galaxy_cli.api_servers[2].validate_certs is (False if ignore_certs_cli else not ignore_certs_cfg if ignore_certs_cfg is not None else True)
```

Comment:
Maybe this would be easier to grasp?
```suggestion
    assert galaxy_cli.api_servers[2].validate_certs is (
        not ignore_certs_cli or (
            ignore_certs_cfg is not None and
            not ignore_certs_cfg
        )
    )
```

Reply:
Alternatively, consider making multiple checks:
```suggestion
    api_server2_validate_certs = galaxy_cli.api_servers[2].validate_certs
    if ignore_certs_cli:
        assert api_server2_validate_certs is False
    elif ignore_certs_cfg is not None:
        assert api_server2_validate_certs is not ignore_certs_cfg
    else:
        assert api_server2_validate_certs is True
```

Reply:
Perhaps, this logic shouldn't even be here in the first place. It is usually discouraged in tests — they should be simple to follow...

Reply:
fixed